app/controllers/api/v1/books_controller.rb

```diff
@@ -7,8 +7,15 @@ def index
 
       def show
         # Not a strong parameter, because :id is part of the API URL
-        @book = Book.find(params[:id])
-        render json: @book
+        respond_to do |format|
+          if Book.exists? params[:id]
+            format.json { render json: Book.find(params[:id]) }
```

Comment:
Alternative idea: catch the exception that the `.find` raises when it is not found in the ApiController, that way you have error handling for every `.find` in every controller and you have a much cleaner `show` endpoint.
Something like

api_controller.rb:
`rescue_from ActiveRecord::RecordNotFound, with: :record_not_found`

def record_not_found
&nbsp;    render json: 'whatever your message is', status: :not_found
end